ifrs17/Report/ReportConfigurationAndUtils.ipynb

```diff
@@ -225,6 +228,20 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static IDataCube<ReportVariable> SelectProjections(this IDataCube<ReportVariable> source, ",
+                "\n                                                            int projectionLevel, ",
+                "\n                                                            ProjectionConfiguration[] orderedProjectionConfigurations) => source.Filter(x => {",
+                "\n    var relevantProjectionConfiguration = orderedProjectionConfigurations.Where((_, i) => i <= projectionLevel).Select(y => y.SystemName);",
```

Comment:
bring relevantProjectionConfiguration  outside the select as it doesn't need to be re-evaluated for each x